app/views/common/_discourse_splash.html.erb

```diff
@@ -93,5 +95,56 @@
       }
     </style>
   </noscript>
+
+  <script nonce="<%= ApplicationHelper.splash_screen_nonce %>">
+    const connectStart = performance?.timing?.connectStart || 0;
+
+    const SPLASH_DELAY = connectStart ? 2000 : 0;
+    const POLLING_INTERVAL = 50;
+
+    const targetTime = connectStart + SPLASH_DELAY;
+    const splashWrapper = document.querySelector("#d-splash");
+    const image = splashWrapper?.querySelector(".preloader-image");
```

Comment:
This suggests that `image` can be undefined, but in `swapSplash()` you access `image.src` unconditionally.